### PR TITLE
Revise Models

### PR DESCRIPTION
**Pull request by LimpidCrypto**
## High Level Overview of Change

This PR makes use of the XRPL `CommonFields` in the models making it easier to work with them.
It also adds some smaller features like:
- `LedgerObject` trait

### Context of Change

As I was working on #71 the current models made it really hard to get model values (mostly from common fields). I had to write workarounds to get/set values of a model making it really inconvenient to work with them. This PR add `CommonFields` introducing a way easier way to work with the models.

### Type of Change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Test Plan

The tests already exist.

## TODO

- [x] fix tests (mostly json string keys in false order)
- [x] recheck field order and if all `common_field` gets `#[serde(flatten)]`
- [x] check if all fields are `pub`
- [x] Add custom field documentation
- [x] run `cargo fix` etc.
- [x] delete file .DS_Store

### CODE REVIEW

File: src/_serde/mod.rs [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 use serde::{de, ser, Deserialize, Deserializer, Serialize, Serializer};
 use serde_json::Value;
 use strum::IntoEnumIterator;
 
 pub type HashMap<K, V> = hashbrown::HashMap<K, V, BuildHasherDefault<FnvHasher>>;
 
-fn serialize_flag<F, S>(flags: &Vec<F>, s: S) -> Result<S::Ok, S::Error>
+pub fn default_none<T>() -> Option<T> {
+    None
+}
```
Comment (sephynox) [inline src/_serde/mod.rs:17]: What's the point of this instead of the Default trait?

File: src/_serde/mod.rs [hunk 2]
```diff
@@ -35,40 +40,46 @@ where
...
+    //                 Err(_) => {
+    //                     return Err(de::Error::custom("SerdeIntermediateStepError: Failed to turn flag into `u32` during deserialization"));
+    //                 }
+    //             };
+    //         }
+    //         Err(_) => {
+    //             return Err(de::Error::custom("SerdeIntermediateStepError: Failed to turn flag into `String` during deserialization"));
+    //         }
+    //     };
+    // }
```
Comment (sephynox) [inline src/_serde/mod.rs:80]: Do we want to keep commented code? I don't think we should.

File: src/models/.DS_Store
Comment (sephynox) [inline src/models/.DS_Store:1]: We don't want to commit this.

File: src/models/mod.rs [hunk 3]
```diff
@@ -49,6 +55,11 @@ pub enum AccountObjectType {
     Ticket,
 }
 
+#[derive(
+    Debug, Eq, PartialEq, Clone, Serialize, Deserialize, Display, AsRefStr, EnumIter, Copy,
+)]
+pub enum NoFlags {}
```
Comment (sephynox) [inline src/models/mod.rs:61]: You should comment this.

File: src/models/transactions/account_set.rs [hunk 4]
```diff
@@ -388,17 +282,15 @@ impl<'a> AccountSetError for AccountSet<'a> {
 impl<'a> AccountSet<'a> {
     pub fn new(
```
Comment (sephynox) [inline src/models/transactions/account_set.rs:283]: Not sure there is a point to these `new` for the sake of a new.

File: src/models/transactions/check_cash.rs [hunk 5]
```diff
@@ -242,40 +137,47 @@ mod test_serde {
...
             None,
             None,
             None,
             None,
             None,
-            Some(Amount::XRPAmount(XRPAmount::from("100000000"))),
+            "838766BA2B995C00744175F69A1B11E32C3DBC40E64801A4056FCBD657F57334".into(),
+            Some("100000000".into()),
             None,
         );
```
Comment (sephynox) [inline src/models/transactions/check_cash.rs:171]: It is harder to understand what is happening with a `new` like this versus just constructing the struct.

**Review by sephynox — COMMENTED** [review]
First pass.

File: src/models/transactions/check_cash.rs (hunk 5 above)
Comment (LimpidCrypto) [reply src/models/transactions/check_cash.rs:171]: So you're saying I should remove all new methods? I get your point, just want to make sure before refactoring.

**Pull request merged**